### DISCUSSION TITLE
Component "<component_name>" not found. Available components are: []

### DISCUSSION THREAD

**Discussion by m5no0p** [tetra-framework/tetra#84] [Q&A] [answered]
I have this error in my browser:
Component "tags_list" not found. Available components are: []

![Screenshot_5](https://github.com/user-attachments/assets/2b4046b3-dba4-4ce8-9204-970888b09ba7)

and my component is like this:

```py 

from tetra import Component, public, Library
from todo.models import Tag

default = Library()

@default.register
class TagsList(Component):
    name = public("")
    color = public("indigo")

    def load(self):
        self.tags = Tag.objects.all()
        
    @public
    def add_todo(self, name, color):
        tag = Tag(
            name=name,
            color=color,
        )
        tag.save()
        self.name = ""
        
    template_name = "todo/components/multi-select-or-create.html"
```

**Comment by nerdoc** [comment]
At the first glance, as there are "no" components available (empty list) it seems to me that your "components" module is not loaded. could it be that there is a syntax error anywhere in your `components.py` file, so that the automatic import does not succeed?
Then NO components are available.
Is your app (that contains the components.py) included in INSTALLED_APPS?
Is this your complete components.py?

**Comment by nerdoc** [comment]
@m5no0p is this still a problem? Or can I close this issue?

**Comment by m5no0p** [comment, marked as answer]
it was my bad...i didn't install sourcetypes package.

![Screenshot_10](https://github.com/user-attachments/assets/69b91185-96f7-4d66-81a9-ea5dc435dab2)
![Screenshot_11](https://github.com/user-attachments/assets/0dc5b2e3-a562-43c8-a50a-10a0313ca225)

**Comment by nerdoc** [comment]
Glad that it worked. `sourcetypes` is not really an obligate requirement of tetra. It just helps you / your IDE to recognize that type of language. So I didn't make tetra depend on it.
